src/corefx/System.Globalization.Native/locale.cpp

```diff
@@ -0,0 +1,373 @@
+			status = GetLocaleInfoDecimalFormatSymbol(locale, DecimalFormatSymbols::kGroupingSeparatorSymbol, value, valueLength);
+			break;
+		case DecimalSeparator:
+			status = GetLocaleInfoDecimalFormatSymbol(locale, DecimalFormatSymbols::kDecimalSeparatorSymbol, value, valueLength);
+			break;
+		case Digits:
+			status = GetDigitSymbol(locale, status, DecimalFormatSymbols::kZeroDigitSymbol, 0, value, valueLength);
+			// symbols kOneDigitSymbol to kNineDigitSymbol are contiguous
+			for (int32_t symbol = DecimalFormatSymbols::kOneDigitSymbol; symbol <= DecimalFormatSymbols::kNineDigitSymbol; symbol++)
+			{
+				int charIndex = symbol - DecimalFormatSymbols::kOneDigitSymbol + 1;
+				status = GetDigitSymbol(locale, status, (DecimalFormatSymbols::ENumberFormatSymbol)symbol, charIndex, value, valueLength);
+			}
+			break;
+		case MonetarySymbol:
+			status = GetLocaleInfoDecimalFormatSymbol(locale, DecimalFormatSymbols::kCurrencySymbol, value, valueLength);
+			break;
+		case Iso4217MonetarySymbol:
+			status = GetLocaleInfoDecimalFormatSymbol(locale, DecimalFormatSymbols::kIntlCurrencySymbol, value, valueLength);
+			break;
+		case MonetaryDecimalSeparator:
+			status = GetLocaleInfoDecimalFormatSymbol(locale, DecimalFormatSymbols::kMonetarySeparatorSymbol, value, valueLength);
+			break;
+		case MonetaryThousandSeparator:
+			status = GetLocaleInfoDecimalFormatSymbol(locale, DecimalFormatSymbols::kMonetaryGroupingSeparatorSymbol, value, valueLength);
+			break;
+		case AMDesignator:
+			status = GetLocaleInfoAmPm(locale, true, value, valueLength);
+			break;
+		case PMDesignator:
+			status = GetLocaleInfoAmPm(locale, false, value, valueLength);
+			break;
+		case PositiveSign:
+			status = GetLocaleInfoDecimalFormatSymbol(locale, DecimalFormatSymbols::kPlusSignSymbol, value, valueLength);
+			break;
+		case NegativeSign:
+			status = GetLocaleInfoDecimalFormatSymbol(locale, DecimalFormatSymbols::kMinusSignSymbol, value, valueLength);
+			break;
+		case Iso639LanguageName:
+			status = CopyCharToUChar(locale.getLanguage(), value, valueLength);
+			break;
+		case Iso3166CountryName:
+			// coreclr expects 2-character version, not 3 (3 would correspond to LOCALE_SISO3166CTRYNAME2 and locale.getISO3Country)
+			status = CopyCharToUChar(locale.getCountry(), value, valueLength);
+			break;
+		case NaNSymbol:
+			status = GetLocaleInfoDecimalFormatSymbol(locale, DecimalFormatSymbols::kNaNSymbol, value, valueLength);
+			break;
+		case PositiveInfinitySymbol:
+			status = GetLocaleInfoDecimalFormatSymbol(locale, DecimalFormatSymbols::kInfinitySymbol, value, valueLength);
+			break;
+		case PercentSymbol:
+			status = GetLocaleInfoDecimalFormatSymbol(locale, DecimalFormatSymbols::kPercentSymbol, value, valueLength);
+			break;
+		case PerMilleSymbol:
+			status = GetLocaleInfoDecimalFormatSymbol(locale, DecimalFormatSymbols::kPerMillSymbol, value, valueLength);
+			break;
+		default:
+			status = U_UNSUPPORTED_ERROR;
+			break;
+	};
```

Comment:
I see now that the managed code passes in a buffer that it expects is always the correct size.  Can we assert something somewhere in here to validate that we didn't get a U_BUFFER_OVERFLOW_ERROR?